ykrt/src/mt.rs

```diff
@@ -378,15 +464,18 @@ mod tests {
         for i in 0..hot_thrsh {
             mtt.control_point(Some(&lp), dummy_step, &mut io);
             assert_eq!(
-                lp.pack.load(Ordering::Relaxed),
+                lp.state.load(Ordering::Relaxed),
                 PHASE_COUNTING | ((i + 1) << PHASE_NUM_BITS)
             );
         }
-        assert_eq!(lp.pack.load(Ordering::Relaxed) & PHASE_TAG, PHASE_COUNTING);
+        assert_eq!(lp.state.load(Ordering::Relaxed) & PHASE_TAG, PHASE_COUNTING);
         mtt.control_point(Some(&lp), dummy_step, &mut io);
-        assert_eq!(lp.pack.load(Ordering::Relaxed) & PHASE_TAG, PHASE_TRACING);
+        assert_eq!(lp.state.load(Ordering::Relaxed) & PHASE_TAG, PHASE_TRACING);
         mtt.control_point(Some(&lp), dummy_step, &mut io);
-        assert_eq!(lp.pack.load(Ordering::Relaxed) & PHASE_TAG, PHASE_COMPILED);
+        assert_eq!(
+            lp.state.load(Ordering::Relaxed) & PHASE_TAG,
+            PHASE_COMPILING
```

Comment:
Could PHASE_COMPILED be observed here too?

Reply:
Good point! Fixed in https://github.com/softdevteam/yk/pull/171/commits/a12aac14e0784954d3adb9d350c134e0726c6e4f.